src/base/utils/misc.h

```diff
@@ -36,7 +36,7 @@
 #include <Windows.h>
 #endif
 
-class QString;
+#include <QString>
```

Comment:
why do you need to change this? please provide full information about it.

Reply:
`loadwinapi` fails to compile on windows-msys2 because of lack of `QString` definition

Reply:
Though it only fails with cmake, works with qmake.

Reply:
by full information, the compilation log is included...

Reply:
```
[1/50] Automatic MOC for target qtsingleapplication
[2/48] Automatic MOC for target qbt_powermanagement
[3/46] Automatic MOC for target qbt_base
[4/45] Building CXX object src/base/CMakeFiles/qbt_base.dir/utils/misc.cpp.obj
FAILED: src/base/CMakeFiles/qbt_base.dir/utils/misc.cpp.obj
E:\Cpp\msys2\mingw64\bin\c++.exe  -DNOMINMAX -DNTDDI_VERSION=0x05010000 -DQBT_VERSION=\"v4.2.0alpha\" -DQBT_VERSION_2=\"4.2.0alpha\" -DQBT_VERSION_BUGFIX=0 -DQBT_VERSION_BUILD=0 -DQBT_VERSION_MAJOR=4 -DQBT_VERSION_MINOR=2 -DQT_CORE_LIB -DQT_DBUS_LIB -DQT_DEPRECATED_WARNINGS -DQT_GUI_LIB -DQT_NETWORK_LIB -DQT_NO_CAST_TO_ASCII -DQT_NO_DEBUG -DQT_NO_DEBUG_OUTPUT -DQT_STRICT_ITERATORS -DQT_USE_QSTRINGBUILDER -DQT_WIDGETS_LIB -DQT_XML_LIB -DSTACKTRACE -DUNICODE -DWIN32 -DWIN32_LEAN_AND_MEAN -D_CRT_SECURE_NO_DEPRECATE -D_SCL_SECURE_NO_DEPRECATE -D_UNICODE -D_WIN32 -D_WIN32_IE=0x0501 -D_WIN32_WINNT=0x0501 -Isrc/base/qbt_base_autogen/include -I../src -isystem E:/Cpp/Sources/OpenSSL-Win64/include -isystem E:/Cpp/Sources/boost/boost_1_70_0 -isystem E:/Cpp/msys2/mingw64/include/QtCore -isystem E:/Cpp/msys2/mingw64/share/qt5/mkspecs/win32-g++ -isystem E:/Cpp/msys2/mingw64/include/QtNetwork -isystem E:/Cpp/msys2/mingw64/include/QtXml -isystem E:/Cpp/msys2/mingw64/include/QtGui -isystem E:/Cpp/msys2/mingw64/include/QtWidgets -isystem E:/Cpp/msys2/mingw64/include/QtDBus -Wall -Wextra -Wfloat-equal -Wcast-qual -Wcast-align -Wsign-conversion -Winvalid-pch -Wno-long-long -fexceptions -frtti -Woverloaded-virtual -Wold-style-cast -Wnon-virtual-dtor -Wfloat-equal -Wcast-qual -Wcast-align -pedantic -pedantic-errors -Wstrict-null-sentinel   -fexceptions -std=c++14 -DBOOST_ASIO_ENABLE_CANCELIO -DWIN32_LEAN_AND_MEAN -DTORRENT_USE_ICONV -DTORRENT_USE_OPENSSL -DTORRENT_USE_LIBCRYPTO -IC:/building/msys64/mingw64/include -std=gnu++14 -MD -MT src/base/CMakeFiles/qbt_base.dir/utils/misc.cpp.obj -MF src\base\CMakeFiles\qbt_base.dir\utils\misc.cpp.obj.d -o src/base/CMakeFiles/qbt_base.dir/utils/misc.cpp.obj -c ../src/base/utils/misc.cpp
In file included from ../src/base/utils/misc.cpp:29:
../src/base/utils/misc.h: In function 'T Utils::Misc::loadWinAPI(const QString&, const char*)':
../src/base/utils/misc.h:96:21: error: 'path' has incomplete type
   96 |             QString path = windowsSystemPath();
      |                     ^~~~
In file included from E:/Cpp/msys2/mingw64/include/QtCore/QtGlobal:1,
                 from ../src/base/utils/misc.h:32,
                 from ../src/base/utils/misc.cpp:29:
E:/Cpp/msys2/mingw64/include/QtCore/qglobal.h:768:7: note: forward declaration of 'class QString'
  768 | class QString;
      |       ^~~~~~~
In file included from ../src/base/utils/misc.cpp:29:
../src/base/utils/misc.h:98:22: error: no match for 'operator+=' (operand types are 'QString' and 'char')
   98 |                 path += '\\';
      |                 ~~~~~^~~~~~~
../src/base/utils/misc.h:100:18: error: no match for 'operator+=' (operand types are 'QString' and 'const QString')
  100 |             path += source;
      |             ~~~~~^~~~~~~~~
../src/base/utils/misc.h:96:21: error: 'path' has incomplete type
   96 |             QString path = windowsSystemPath();
      |                     ^~~~
In file included from E:/Cpp/msys2/mingw64/include/QtCore/QtGlobal:1,
                 from ../src/base/utils/misc.h:32,
                 from ../src/base/utils/misc.cpp:29:
E:/Cpp/msys2/mingw64/include/QtCore/qglobal.h:768:7: note: forward declaration of 'class QString'
  768 | class QString;
      |       ^~~~~~~
In file included from ../src/base/utils/misc.cpp:29:
../src/base/utils/misc.h:96:21: error: 'path' has incomplete type
   96 |             QString path = windowsSystemPath();
      |                     ^~~~
In file included from E:/Cpp/msys2/mingw64/include/QtCore/QtGlobal:1,
                 from ../src/base/utils/misc.h:32,
                 from ../src/base/utils/misc.cpp:29:
E:/Cpp/msys2/mingw64/include/QtCore/qglobal.h:768:7: note: forward declaration of 'class QString'
  768 | class QString;
      |       ^~~~~~~
In file included from ../src/base/utils/misc.cpp:29:
../src/base/utils/misc.h: In instantiation of 'T Utils::Misc::loadWinAPI(const QString&, const char*) [with T = unsigned char (*)(unsigned char, unsigned char, unsigned char)]':
../src/base/utils/misc.cpp:121:109:   required from here
../src/base/utils/misc.h:105:20: warning: cast between incompatible function types from 'FARPROC' {aka 'long long int (*)()'} to 'unsigned char (*)(unsigned char, unsigned char, unsigned char)' [-Wcast-function-type]
  105 |             return reinterpret_cast<T>(
      |                    ^~~~~~~~~~~~~~~~~~~~
  106 |                 ::GetProcAddress(::LoadLibraryW(pathWchar.get()), funcName));
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
ninja: build stopped: subcommand failed.
```

Reply:
ok?